py/desispec/night_qa.py

```diff
@@ -0,0 +1,740 @@
+        outdir: output folder name (string)
+        night: night (int)
+
+    Returns:
+        dictionary with file names
+    """
+    return {
+        "html" : os.path.join(outdir, "nightqa-{}.html".format(night)),
+        "dark" : os.path.join(outdir, "dark-{}.mp4".format(night)),
+        "badcol" : os.path.join(outdir, "badcol-{}.mp4".format(night)),
+        "ctedet" : os.path.join(outdir, "ctedet-{}.mp4".format(night)),
+        "sframesky" : os.path.join(outdir, "sframesky-{}.mp4".format(night)),
+        "tileqa" : os.path.join(outdir, "tileqa-{}.mp4".format(night)),
+    }
+
+
+
+def get_survey_night_expids(
+    night,
+    survey,
+    datadir = os.getenv("DESI_SPECTRO_DATA")):
+    """
+    TBD
+    """
+    fns = sorted(
+        glob(
+            os.path.join(
+                datadir,
+                "{}".format(night),
+                "????????",
+                "desi-????????.fits.fz",
+            )
+        )
+    )
+    expids, tileids = [], []
+    for i in range(len(fns)):
+        hdr = fits.getheader(fns[i], "SPEC")
+        if hdr["OBSTYPE"] == "SCIENCE":
+            if hdr["NTSSURVY"] == survey:
+                expids.append(hdr["EXPID"])
+                tileids.append(hdr["TILEID"])
+    log.info(
+        "found {} exposures from {} tiles for SURVEY={} and NIGHT={}".format(
+            len(expids), np.unique(tileids).size, survey, night,
+        )
+    )
+    return np.array(expids), np.array(tileids)
+
+
+def get_dark_night_expid(night, datadir = os.getenv("DESI_SPECTRO_DATA")):
+    """
+    TBD
+    """
+    fns = sorted(
+        glob(
+            os.path.join(
+                datadir,
+                "{}".format(night),
+                "????????",
+                "desi-????????.fits.fz",
+            )
+        )
+    )
+    expid = None
+    for i in range(len(fns)):
+        hdr = fits.getheader(fns[i], "SPEC")
+        if (hdr["OBSTYPE"] == "DARK") & (hdr["REQTIME"] == 300):
+            expid = hdr["EXPID"]
+            break
+    if expid is None:
+        log.warning(
+            "no EXPID found as the 300s DARK for NIGHT={}".format(night)
+        )
+    else:
+        log.info(
+            "found EXPID={} as the 300s DARK for NIGHT={}".format(
+                expid, night,
+            )
+        )
+    return expid
+
+
+def create_mp4(fns, outmp4, duration=15):
+    """
+    Create an animated .mp4 from a set of input files (usually pngs).
+
+    Args:
+        fns: list of input filenames, in the correct order (list of strings)
+        outmp4: output .mp4 filename
+        duration (optional, defaults to 15): video duration in seconds (float)
+
+    Notes:
+        Requires ffmpeg to be installed.
+        At NERSC, run in the bash command line: "module load ffmpeg".
+        The movie uses fns in the provided order.
+    """
+    # AR is ffmpeg installed
+    if os.system("which ffmpeg") != 0:
+        log.error("ffmpeg needs to be installed to run create_mp4(); exiting")
```

Comment:
Please add a message like 'at NERSC, this is available with "module load ffmpeg" '.  Please also do this check earlier in the code so that it doesn't run for several minutes and then exit due to a missing module).

Reply:
done